python/pyarrow/tests/parquet/test_metadata.py

```diff
@@ -794,3 +794,28 @@ def test_column_chunk_key_value_metadata(parquet_test_datadir):
     assert key_value_metadata1 == {b'foo': b'bar', b'thisiskeywithoutvalue': b''}
     key_value_metadata2 = metadata.row_group(0).column(1).metadata
     assert key_value_metadata2 is None
+
+
+def test_internal_class_instantiation():
+    def msg(c):
+        return f"Can't instantiate internal class {c}"
+
+    with pytest.raises(TypeError) as excinfo:
+        pq.Statistics()
+        assert excinfo.value == msg("Statistics")
```

Comment:
For the record, you could also use the `match` parameter. Though checking `excinfo.value` is fine as well.
```suggestion
    with pytest.raises(TypeError, match=msg("Statistics")) as excinfo:
        pq.Statistics()
```